### PR TITLE
adding enclave id to response

### PR DESCRIPTION
**Pull request by bvavala**
Signed-off-by: Bruno Vavala <bruno.vavala@intel.com>

### CODE REVIEW

**Review by g2flyer — APPROVED** [review]
Looks good and works as expected (i.e., runs now past the credentials retrieval but then fails in signature verification).   Two additional inline comments: one is not directly relevant to this PR. The other one would be a nice-to-have but not essential. As you have anyway to rebase: just decide whether you want to address it or not and either way just merge once done?

File: internal/utils/proto.go [hunk 1]
```diff
@@ -43,8 +44,10 @@ func UnmarshalCredentials(credentialsBase64 string) (*protos.Credentials, error)
 
 // returns enclave_id as hex-encoded string of SHA256 hash over enclave_vk.
 func GetEnclaveId(attestedData *protos.AttestedData) string {
+	// hash enclave vk
 	h := sha256.Sum256(attestedData.EnclaveVk)
-	return hex.EncodeToString(h[:])
+	// encode and normalize
+	return strings.ToLower(hex.EncodeToString(h[:]))
```
Comment (g2flyer) [inline internal/utils/proto.go:50]: `ecc/chaincode/enclave/mock_enclave.go` has also one of these which you will have to change (but as mentioned above, for consistency probably better to leave this here as-is and change c-code to `toupper`?

File: common/enclave/cc_data.cpp [hunk 2]
```diff
@@ -42,8 +42,11 @@ bool cc_data::generate_keys()
         state_encryption_key_ = pdo::crypto::skenc::GenerateKey();
 
         // debug
```
[the remark this answers is not in the document]
Comment (bvavala) [reply common/enclave/cc_data.cpp]: done

File: common/enclave/cc_data.cpp [hunk 3]
```diff
@@ -245,3 +248,18 @@ ByteArray cc_data::get_state_encryption_key()
...
+{
+    // get enclave vk
+    std::string s = verification_key_.Serialize();
+    // hash
+    ByteArray h = pdo::crypto::ComputeMessageHash(ByteArray(s.c_str(), s.c_str() + s.length()));
+    // hex encode
+    std::string hex = ByteArrayToHexEncodedString(h);
+    // normalize
+    std::transform(
+        hex.begin(), hex.end(), hex.begin(), [](unsigned char c) { return std::tolower(c); });
```
[the remark this answers is not in the document]
Comment (bvavala) [reply common/enclave/cc_data.cpp]: no strong opinion, toupper is fine

File: internal/utils/validation.go [hunk 4]
```diff
@@ -73,6 +80,10 @@ func ReplayReadWrites(stub shim.ChaincodeStubInterface, rwset *kvrwset.KVRWSet)
 }
 
 func Validate(responseMsg *protos.ChaincodeResponseMessage, readset, writeset [][]byte, attestedData *protos.AttestedData) error {
+	if responseMsg.Signature == nil {
+		return fmt.Errorf("absent enclave signature")
+	}
+
 	// Note: below signature was created in ecc_enclave/enclave/enclave.cpp::gen_response
 	hash := ComputedHash(responseMsg, readset, writeset)
```
[the remark this answers is not in the document]
Comment (bvavala) [reply internal/utils/validation.go:88]: > BTW: as adding signature i guess is your next task and you will have to touch ComputedHash. Could you maybe also rename it with a bit more meaningful name such as ComputeEndorsementSignatureMessage or alike (something which makes it clear it is not an general hash function it is really the payload for the enclave endorsement signature).

yes as I get there

> BTW(BTW): given that signature algorithm already hashes message, we might also just concatenate and drop the (double) hashing inside this function?

yes as I get there

> PS: ecc/chaincode/enclave/mock_enclave.go also has a reference to the function which you might not notice by testing due to gotags not enabling it by defaults ...

done

File: internal/utils/proto.go (hunk 1 above)
Comment (bvavala) [reply internal/utils/proto.go]: ported both to toupper

**Pull request merged**